File: backend/app/Models/Lab.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Lab extends Model
{
    // Primary key for labs table
    protected $primaryKey = 'lab_id';

    // Fields allowed for mass assignment
    protected $fillable = [

        'section_id',
        'lab_name',
        'capacity',
        'enrolled',
        'schedule_day',
        'schedule_time',
    ];

    // Relationship: Lab belongs to a section
    public function section()
    {
        return $this->belongsTo(
            Section::class,
            'section_id',
            'section_id'
        );
    }

    // Relationship: Lab has many registrations
    public function registrations()
    {
        return $this->hasMany(
            Registration::class,
            'lab_id',
            'lab_id'
        );
    }
}

File: frontend/lib/domain/lab.dart
class LabModel {

  final int labId;

  final String labName;

  final int capacity;

  final int enrolled;

  final String scheduleDay;

  final String scheduleTime;

  LabModel({

    required this.labId,

    required this.labName,

    required this.capacity,

    required this.enrolled,

    required this.scheduleDay,

    required this.scheduleTime,
  });

  factory LabModel.fromJson(
      Map<String, dynamic> json) {

    return LabModel(

      labId: json['lab_id'] ?? 0,

      labName:
          json['lab_name'] ?? '',

      capacity:
          json['capacity'] ?? 0,

      enrolled:
          json['enrolled'] ?? 0,

      scheduleDay:
          json['schedule_day'] ?? '',

      scheduleTime:
          json['schedule_time'] ?? '',
    );
  }
}